util/xrh_header.go

```diff
@@ -0,0 +1,47 @@
+package util
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	"github.com/RedHatInsights/sources-api-go/kafka"
+	"github.com/redhatinsights/platform-go-middlewares/identity"
+)
+
+func ParseXRHIDHeader(inputIdentity string) (*identity.XRHID, error) {
+	var XRHIdentity *identity.XRHID
+	decodedIdentity, err := base64.StdEncoding.DecodeString(inputIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding Identity: %v", err)
+	}
+
+	err = json.Unmarshal(decodedIdentity, &XRHIdentity)
+	if err != nil {
+		return nil, fmt.Errorf("x-rh-identity header does not contain valid JSON")
+	}
+
+	return XRHIdentity, nil
+}
+
+func AccountNumberFromHeaders(headers []kafka.Header) (string, error) {
+	XRHAccountNumberKey := "x-rh-sources-account-number"
+	XRHIdentityKey := "x-rh-identity"
```

Comment:
I moved them from function scope, thanks 

middleware: What's reason to extracting them to middleware ? - is it because we can use them also in middleware ? 
Anyway let's do it in follow up PR.

Reply:
Yes - `middleware/headers.go` is where they get parsed into the context. Thinking about moving that to its own package `middleware/headers/`.  

Either way we can do it in a followup PR.